packages/theme-language-server-common/src/renamed/handlers/AssetRenameHandler.ts

```diff
@@ -27,18 +26,29 @@ import { BaseRenameHandler } from '../BaseRenameHandler';
  * WorkspaceEdit that changes the references to the new asset.
  */
 export class AssetRenameHandler implements BaseRenameHandler {
-  constructor(private connection: Connection, private capabilities: ClientCapabilities) {}
+  constructor(
+    private documentManager: DocumentManager,
+    private connection: Connection,
+    private capabilities: ClientCapabilities,
+    private findThemeRootURI: (uri: string) => Promise<string>,
+  ) {}
 
-  async onDidRenameFiles(params: RenameFilesParams, theme: AugmentedSourceCode[]): Promise<void> {
+  async onDidRenameFiles(params: RenameFilesParams): Promise<void> {
     if (!this.capabilities.hasApplyEditSupport) return;
     const isLiquidSourceCode = (file: AugmentedSourceCode): file is AugmentedLiquidSourceCode =>
       file.type === SourceCodeType.LiquidHtml;
 
-    const liquidSourceCodes = theme.filter(isLiquidSourceCode);
     const relevantRenames = params.files.filter(
       (file) => isAsset(file.oldUri) && isAsset(file.newUri),
     );
 
+    // Only preload if you have something to do
+    if (relevantRenames.length === 0) return;
+    const rootUri = await this.findThemeRootURI(path.dirname(params.files[0].oldUri));
+    await this.documentManager.preload(rootUri);
```

Comment:
Also further gated the preload behind the hasApplyEdit support here. We'll only preload if we know we must preload.